src/Runtime/LoopIterator.php

```diff
@@ -34,7 +34,18 @@ final class LoopIterator implements \Iterator
 
     public function __construct($seq)
     {
-        $this->seq = is_iterable($seq) ? (is_array($seq) ? new \ArrayIterator($seq) : $seq) : new \ArrayIterator([]);
+        if (is_array($seq)) {
+            $this->seq = new \ArrayIterator($seq);
+        } elseif ($seq instanceof \IteratorAggregate) {
+            do {
+                $seq = $seq->getIterator();
+            } while ($seq instanceof \IteratorAggregate);
+            $this->seq = $seq;
```

Comment:
PHPStorm was not happy so i looked a bit and...

 [IteratorAggregate::getIterator()](https://www.php.net/manual/en/iteratoraggregate.getiterator.php) does not return an Iterator but only a Traversable ... 
 
 so after the loop here $this->seq could totally be a simple array
 
 
Would something like this work 'out of the box' ? 

```
if (is_array($seq))
 // ArrayIterator($seq)
elseif ($seq instanceof Traversable)
  // IteratorIterator ($seq)
else 
  // EmptyIterator

```

Reply:
The best would be to create a PR with examples of something that fails and the fix for those tests.

Reply:
Well i spend time and clearly IRL that won't ever happen. 

The last possibilities would be those classes PHP say can be used in "foreach" while not implementing "Iterator" directly, but i cannot find an example.

Sorry for the waste of time here.